### PR TITLE
git-annex-remote-rclone 0.6 (new formula)

### PR DESCRIPTION
**Pull request by dankessler**
First time contributing to homebrew-core. I've tried very hard to review all of your (very well-written) guidelines, but I apologize in advance if I've overlooked anything

Add formula for git-annex-remote-rclone. This tool enables git-annex
to use any of the providers supported by rclone as a special remote
when using git-annex. This is available for debian via apt, but is not
currently in homebrew. Installation is as simple as copying a bash
script onto the path. Writing a test was tricky, since using
git-annex-remote-rclone requires creating a new special remote using
an extant rclone target, so following the guidance in the formula
cookbook, I've tried to initialize a remote using target
testtest123123, which hopefully nobody uses as a name for a remote in
rclone. If git-annex-remote-rclone is correctly installed, it will
generate a specific error message which the test checks.

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by jonchang** [comment]
> Installation is as simple as copying a bash  script onto the path.

This should be `bottle :unneeded` then.

**Comment by dankessler** [comment]
@jonchang does this prevent the construction of what would be a trivial bottle? Are there any other changes I'd need to make?

**Comment by dankessler** [comment]
When I fix this, is it preferred that I squash my new changes into the previous commit and force the push? Or just add a commit to the PR? I’m not clear on how strictly you want to adhere to the 1 commit : 1 formula rule from the [cookbook](https://docs.brew.sh/Formula-Cookbook#push)

**Comment by dankessler** [comment]
I'm going to use this opportunity to also try to write a better test if I can. I'll make additional commits under the assumption that I can always squash them later if requested.

**Comment by SMillerDev** [comment]
Eventually it'll need to be squashed before it can be merged.

**Comment by dankessler** [comment]
I've now written a substantially more robust test; details documented in commit message https://github.com/Homebrew/homebrew-core/pull/49468/commits/594a40bfd56b76f56248347f3d5996c51da17a2d. I think things are ready to go / be reviewed now.

For now I've left things in 3 separate commits, but will happily squash and do a force push on request, but I've never done a force push with a PR so am not clear what the consequences will be and am holding off until asked to do so.

**Comment by dankessler** [comment]
Well that was embarrassing; I had a second tap active which had an (audit-passing) version of the same formula, and it seems that this confuses `brew audit` so I missed these issues.

**Comment by dankessler** [comment]
I'm not sure what the code review process is, so I wanted to double check if there was some action I needed to take (perhaps you *do* want me to squash these commits?) in order to get this reviewed and (hopefully) accepted.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Thanks @dankessler for your contribution to Homebrew! :tada: 🥇 

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Comment by chenrui333** [comment]
> I'm not sure what the code review process is, so I wanted to double check if there was some action I needed to take (perhaps you _do_ want me to squash these commits?) in order to get this reviewed and (hopefully) accepted.

I think it looks good, just one squashed commit away.

**Comment by chenrui333** [comment]
> Eventually it'll need to be squashed before it can be merged.

cc @dankessler

**Comment by dankessler** [comment]
@chenrui333 do you prefer I do the squashing, or do you do that when accepting the PR?

**Comment by chenrui333** [comment]
Sorry about so many back and forth label changes. 😅

**Comment by dankessler** [comment]
No problem. So it looks like you have done the squashing, so there's nothing further you need from me, correct?

**Pull request closed without merging**
